store/store.go

```diff
@@ -329,6 +352,10 @@ func (bs *BlockStore) LoadBlockMetaByHash(hash []byte) *types.BlockMeta {
 // and it comes from the block.LastCommit for `height+1`.
 // If no commit is found for the given height, it returns nil.
 func (bs *BlockStore) LoadBlockCommit(height int64) *types.Commit {
+	comm, ok := bs.blockCommitCache.Get(height)
+	if ok {
+		return comm.Clone()
```

Comment:
Do we need to return a clone here? I assume this data never changes...

Reply:
Returning the Clone replicates existing behavior. (A caller mutation would pollute the cache)

We can change the API to remove the clone, which I think is a good idea, we would just have to look for all usages and ensure they don't mutate it

Reply:
@ValarDragon would you prefer changing the API here or in another PR?

Reply:
Another PR